servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -300,28 +300,65 @@ public HttpServerBuilder bufferAllocator(final BufferAllocator allocator) {
+        return listenForService(streamingService, serviceFilters, streamingService.requiredOffloads());
     }
 
     @Override
     public Single<HttpServerContext> listenStreaming(final StreamingHttpService service) {
-        return listenForService(service, computeServiceStrategy(StreamingHttpService.class, service));
+        final List<StreamingHttpServiceFilterFactory> serviceFilters = initServiceFilters();
+        return listenForService(service, serviceFilters, computeServiceStrategy(
+                StreamingHttpService.class, service, this.strategy, serviceFilters));
     }
 
     @Override
     public Single<HttpServerContext> listenBlocking(final BlockingHttpService service) {
-        StreamingHttpService streamingService = toStreamingHttpService(
-                computeServiceStrategy(BlockingHttpService.class, service), service);
-        return listenForService(streamingService, streamingService.requiredOffloads());
+        final List<StreamingHttpServiceFilterFactory> serviceFilters = initServiceFilters();
+        final StreamingHttpService streamingService = toStreamingHttpService(computeServiceStrategy(
+                BlockingHttpService.class, service, this.strategy, serviceFilters), service);
+        return listenForService(streamingService, serviceFilters, streamingService.requiredOffloads());
     }
 
     @Override
     public Single<HttpServerContext> listenBlockingStreaming(final BlockingStreamingHttpService service) {
-        StreamingHttpService streamingService = toStreamingHttpService(
-                computeServiceStrategy(BlockingStreamingHttpService.class, service), service);
-        return listenForService(streamingService, streamingService.requiredOffloads());
+        final List<StreamingHttpServiceFilterFactory> serviceFilters = initServiceFilters();
+        final StreamingHttpService streamingService = toStreamingHttpService(computeServiceStrategy(
+                BlockingStreamingHttpService.class, service, this.strategy, serviceFilters), service);
+        return listenForService(streamingService, serviceFilters, streamingService.requiredOffloads());
+    }
+
+    private List<StreamingHttpServiceFilterFactory> initServiceFilters() {
+        // Take snapshot of currently appended filters:
+        final List<StreamingHttpServiceFilterFactory> filters = new ArrayList<>(this.serviceFilters);
+
+        // Append internal filters:
+        // 1. The watchdog sits at the very beginning of the response flow (the end of the filter pipeline) so that any
+        // payload coming from the service is ensured to be tracked before subsequent filters get a chance to drop it
+        // without being accounted for.
+        filters.add(HttpMessageDiscardWatchdogServiceFilter.INSTANCE);
+
+        return unmodifiableList(filters);
+    }
+
+    private List<StreamingHttpServiceFilterFactory> initNonOffloadsServiceFilters(
```

Comment:
I would suggest renaming this method to something that includes the word `default` so its understood that those are the default ones always added

Reply:
Because both methods do 2 jobs (add default filters and copy user filters to avoid side-effects), I couldn't fit "default" nicely in a new name. "init" is abstract enough, methods are short and have extensive comments about what they do.